Sources/SwiftFrameCore/Config/ScreenshotData.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+
+public struct ScreenshotData: Decodable, ConfigValidatable, Equatable {
+
+    // MARK: - Properties
+
+    let screenshotName: String
+    let bottomLeft: Point
+    let bottomRight: Point
+    let topLeft: Point
+    let topRight: Point
+    private let _zIndex: Int?
```

Comment:
Looking at this again now, do we actually need to store the optioinal `_zIndex`? Or could we not perform the `?? 0` inside the `init` and only store the non-optional index?

Reply:
that's a common issue I've stumbled across working with `Codable` a lot of times. Sometimes I want a value to optional in JSON but not in code and provide a default value. I fiddled around a `propertyWrapper` for this but without much success

Reply:
Ah, you're right. It's no avoidable if you also want to init from JSON. Seems fine the.